src/pulseshapes.jl

```diff
@@ -0,0 +1,102 @@
+module PulseShapes
+# provides unscaled pulse shapes for waveforms
+
+import Base.tanh
+
+"""
+	constant(;pulse_length=0.0, sampling_rate=1.2e9)
+
+Constant pulse shape for delays or square pulses.
+"""
+function constant(;pulse_length=0.0, sampling_rate=1.2e9)
+	num_pts = round(UInt, pulse_length*sampling_rate)
+	ones(Complex128, num_pts)
+end
+
+"""
+	gaussian(;pulse_length=0.0, sampling_rate=1.2e9, cutoff=2.0)::Vector{Complex128}
+
+Gaussian shaped pulse of a given pulse_length and sampling_rate going between -cutoff and cutoff σ wide.
+Shape is shifted down to avoid finite cutoff step and start and end of pulse.
+"""
+function gaussian(;pulse_length=0.0, sampling_rate=1.2e9, cutoff=2.0)::Vector{Complex128}
+	num_pts = round(UInt, pulse_length*sampling_rate)
+	x_pts = linspace(-cutoff, cutoff, num_pts)
+	# calculate the distance between x_points to extrapolate to the next point
+	# and shift that point to zero
+	x_step = x_pts[2] - x_pts[1]
+	next_val = exp(-0.5 * (x_pts[1] - x_step)^2)
+	shape = 1/(1-next_val) .* (exp.(-0.5 * (x_pts.^2)) - next_val)
```

Comment:
So that similar to the gaussOn/gaussOff we go all the way to "amp" or 1.0 in this case. Since it will cause compatibility issues we could comment it out for now.